pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -0,0 +1,120 @@
+package kubevirt
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/kubevirt"
+	"github.com/openshift/installer/pkg/types/kubevirt/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate executes kubevirt specific validation
+func Validate(ic *types.InstallConfig, clientBuilder ClientBuilderFuncType) error {
+	kubevirtPlatformPath := field.NewPath("platform", "kubevirt")
+
+	if ic.Platform.Kubevirt == nil {
+		return errors.New(field.Required(
+			kubevirtPlatformPath,
+			"validation requires a KubeVirt platform configuration").Error())
+	}
+
+	return validatePlatform(ic.Platform.Kubevirt, ic.MachineNetwork, kubevirtPlatformPath, clientBuilder).ToAggregate()
+}
+
+func validatePlatform(kubevirtPlatform *kubevirt.Platform, machineNetworkEntryList []types.MachineNetworkEntry, fldPath *field.Path, clientBuilder ClientBuilderFuncType) field.ErrorList {
+	allErrs := validation.ValidatePlatform(kubevirtPlatform, fldPath)
+
+	client, resultErrs := validateInfraClusterReachable(kubevirtPlatform.Namespace, fldPath, clientBuilder)
+	allErrs = append(allErrs, resultErrs...)
+	if client != nil {
+		allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath)...)
```

Comment:
You should pass in the field path that is being validated rather than relying on the callee to adjust it.
```suggestion
		allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath.Child("storageClass")...)
```

Reply:
Fixed